### PR TITLE
Remove pause feature

### LINKED ISSUE #809: Make pause stable
https://github.com/metalbear-co/mirrord/pull/791 introduces the unstable pause feature for https://github.com/metalbear-co/mirrord/issues/712. Pause should be made stable, then unmarked as unstable.
To become stable what we need is:

- [x] #1357
- [x] #1358
- [x] #1408
- [ ] something else?

### LINKED ISSUE #1103: Wait for container to [un]pause when [un]pausing a container with docker
Check if with docker the pause call returns when the request to pause was made, not when the container has been successfully paused. If that's the case, wait for [un]pause to complete before returning.

Maybe we can wait only on unpause, and that way still prevent inconsistencies when pausing and unpausing multiple times quickly, without slowing down client connections (agent does not wait for container to pause before moving on with handling the new connection).

### LINKED ISSUE #1737: agent pause fails
[2m2023-08-01T09:25:05.962963Z[0m [31mERROR[0m ThreadId(02) [1mmirrord_agent[0m[2m:[0m Error handling client message: Status(Status { code: Unknown, message: "cannot pause a paused container: unknown", metadata: MetadataMap { headers: {"content-type": "application/grpc"} }, source: None })

### LINKED ISSUE #1878: Errors on already paused sometimes
Sometimes users get container already paused error, happened to me when agent failed launching with iptables rules failure (my bug) but seems to maybe reproduce it?

### LINKED ISSUE #2012: pause on ephemeral agent breaks
### Bug Description

It seems that the container is not unpausing after stopping the debug session (using Rider). The first time it works (ephemeral agent + pause), but any subsequent attempt fails

### Steps to Reproduce

1. Run mirrord with pause and ephemeral agent, it should work the first time
2. Run it again, this time it should break and will only work again once the pod has been restarted

### Backtrace

```shell
│ "Timeout requesting for target container pause.",

[2m2023-10-18T11:46:09.106207Z[0m [31mERROR[0m [2mmirrord_kube::api[0m[2m:[0m Daemon sent log message Failed to pause target container: PauseFailedCgroup(CgroupV2Error(NoPidsFoundInCgroup))
```

### Relevant Logs

_No response_

### Your operating system and version

macOS 14

### Local process

dotnet

### Local process version

_No response_

### Additional Info

_No response_

### LINKED ISSUE #2046: agent leaves pod paused
### Bug Description

User gets "already paused" for a containerd based pod.

### Steps to Reproduce

n/a

### Backtrace

_No response_

### Relevant Logs

_No response_

### Your operating system and version

macOS

### Local process

python flask

### Local process version

_No response_

### Additional Info

_No response_

### LINKED ISSUE #2458: Remove pause
Related #1737 #809 #2012 #1878 #2046 #1103

### PR DESCRIPTION
**Pull request by aviramha**
Closes #2458 
Closes #1737 
Closes #809 
Closes #2012
Closes #1878
Closes #2046
Closes #1103

### CODE REVIEW

File: mirrord/agent/src/entrypoint.rs [hunk 1]
```diff
@@ -452,30 +452,10 @@ impl ClientConnectionHandler {
...
-                        .await?;
-                    }
-                    Err(e) => {
-                        self.respond(DaemonMessage::LogMessage(LogMessage::error(format!(
-                            "Failed to pause target container: {e:?}"
-                        ))))
-                        .await?;
-                    }
-                }
+                self.respond(DaemonMessage::LogMessage(
```
Comment (Razz4780) [inline mirrord/agent/src/entrypoint.rs:455]: `DaemonMessage::Close` maybe?

File: mirrord/cli/src/config.rs [hunk 2]
```diff
@@ -153,10 +153,6 @@ pub(super) struct ExecArgs {
     #[arg(long = "steal")]
     pub tcp_steal: bool,
 
-    /// Pause target container while running.
-    #[arg(short, long, alias = "paws")]
```
Comment (Razz4780) [inline mirrord/cli/src/config.rs:157]: The hell was this alias? :laughing:

File: mirrord/protocol/src/codec.rs [hunk 3]
```diff
@@ -141,6 +141,7 @@ pub enum DaemonMessage {
     /// NOTE: can remove `RemoteResult` when we break protocol compatibility.
     GetEnvVarsResponse(RemoteResult<HashMap<String, String>>),
     GetAddrInfoResponse(GetAddrInfoResponse),
+    /// Pause is deprecated but we don't want to break protocol
```
Comment (Razz4780) [inline mirrord/protocol/src/codec.rs:144]: Maybe add [this](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute)?

Comment (aviramha) [reply mirrord/protocol/src/codec.rs:144]: We still want agent to respond with nice log until we break protocol, so I don't think this is helpful.

File: mirrord/agent/src/entrypoint.rs (hunk 1 above)
Comment (aviramha) [reply mirrord/agent/src/entrypoint.rs:455]: It'd make a very weird situation where old cli new agent just closes connection..

File: mirrord/cli/src/config.rs (hunk 2 above)
Comment (aviramha) [reply mirrord/cli/src/config.rs:157]: I think @meowjesty is the genius behind this

File: mirrord/agent/src/entrypoint.rs (hunk 1 above)
Comment (Razz4780) [reply mirrord/agent/src/entrypoint.rs:455]: But with a descriptive message :shrug: 
I think it's better than silently (best case that the log is printed to file by intproxy) ignoring pause request

File: mirrord/protocol/src/codec.rs (hunk 3 above)
Comment (Razz4780) [reply mirrord/protocol/src/codec.rs:144]: Okok

**Review by Razz4780 — APPROVED** [review]

**Pull request merged**